src/os/ui/query/queryhandler.js

```diff
@@ -269,55 +269,103 @@ os.ui.query.QueryHandler.filters = function(item) {
  * @param {!Object<string, string|boolean>} item
  * @return {!ol.Feature}
  */
-os.ui.query.QueryHandler.toExclude = function(item) {
-  return /** @type {!ol.Feature} */ (os.ui.areaManager.get(/** @type {string} */ (item['areaId'])));
+os.ui.query.QueryHandler.prototype.toExclude = function(item) {
+  return this.getArea(/** @type {string} */ (item['areaId']));
 };
 
 
 /**
- * Creates the filter
- *
- * @return {string}
+ * Gets an area from an id
+ * @param {string} id
+ * @return {!ol.Feature}
  */
-os.ui.query.QueryHandler.prototype.createFilter = function() {
-  var qm = os.ui.queryManager;
-  var am = os.ui.areaManager;
-  var fm = os.ui.filterManager;
+os.ui.query.QueryHandler.prototype.getArea = function(id) {
+  return /** @type {!ol.Feature} */ (os.ui.areaManager.get(id));
+};
 
-  // we don't care about layer in the order array
-  var order = ['area', 'filter'];
 
-  var layerId = this.getLayerId();
-  var qmEntries = qm.getEntries(layerId, null, null, true);
+/**
+ * Gets a filter from an id
+ * @param {string} id
+ * @return {!os.filter.FilterEntry}
+ */
+os.ui.query.QueryHandler.prototype.getFilter = function(id) {
+  return /** @type {!os.filter.FilterEntry} */ (os.ui.filterManager.getFilter(id));
+};
+
+
+/**
+ * Gets entries from an id
+ * @param {?string} id
+ * @return {!Array<!Object<string, string|boolean>>}
+ */
+os.ui.query.QueryHandler.prototype.getEntries = function(id) {
+  return os.ui.queryManager.getEntries(id, null, null, true);
+};
+
+
+
+/**
+ * Gets entries for the filter
+ * @param {!Array<!Object<string, string|boolean>>} entries
+ * @param {!Array<!Object<string, string|boolean>>} includes
+ * @param {!Array<!Object<string, string|boolean>>} excludes
+ */
+os.ui.query.QueryHandler.prototype.getActiveEntries = function(entries, includes, excludes) {
```

Comment:
This is a bit disingenuous because it is a getter that doesn't return anything. I'd prefer returning the entries over forcing the array to be created elsewhere.

Reply:
I was on the fence about that too, will fix